addons/binding/org.openhab.binding.paradoxalarm/src/main/java/org/openhab/binding/paradoxalarm/internal/util/ParadoxUtil.java

```diff
@@ -0,0 +1,142 @@
+        printByteArray(description, array, array[1] + 16);
+    }
+
+    public static void printByteArray(String description, byte[] array, int length) {
+        if (description != null && !description.isEmpty()) {
+            logger.trace(description);
+        }
+        int countBytes = 0;
+        String result = "";
+        for (int index = 0; index < length; index++) {
+            countBytes++;
+            String st = String.format("0x%02X,\t", array[index]);
+            result += st;
+            if (countBytes > 7) {
+                logger.trace(result);
+                countBytes = 0;
+                result = "";
+                continue;
+            }
+        }
+        if (!result.isEmpty()) {
+            logger.trace(result);
+        }
+
+    }
+
+    public static byte setBit(byte byteValue, int i, int j) {
+        if (j == 1) {
+            return (byte) (byteValue | (1 << i));
+        } else {
+            return (byte) (byteValue & ~(1 << i));
+        }
+    }
+
+    public static byte getHighNibble(byte value) {
+        return (byte) ((value & 0xF0) >> 4);
+    }
+
+    public static byte getLowNibble(byte value) {
+        return (byte) (value & 0x0F);
+    }
+
+    public static byte[] mergeByteArrays(byte[]... arrays) {
+        try {
+            ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
+            for (byte[] array : arrays) {
+                outputStream.write(array);
+            }
+            byte[] byteArray = outputStream.toByteArray();
+            return byteArray;
+        } catch (IOException e) {
+            logger.warn("Exception merging arrays: {}", e);
+            return new byte[0];
+        }
+    }
+
+    public static byte[] intToByteArray(int value) {
+        return ByteBuffer.allocate(Integer.SIZE / Byte.SIZE).order(ByteOrder.BIG_ENDIAN).putInt(value).array();
+    }
+
+    public static byte[] shortToByteArray(short value) {
+        return ByteBuffer.allocate(Short.SIZE / Byte.SIZE).order(ByteOrder.BIG_ENDIAN).putShort(value).array();
+    }
+
+    public static byte[] stringToBCD(String pcPassword) {
+        return stringToBCD(pcPassword, 4);
+    }
+
+    public static byte[] stringToBCD(String pcPassword, int numberOfDigits) {
+        byte[] result = new byte[numberOfDigits / 2];
+        for (int i = 0, j = 0; i < 2; i++, j += 2) {
+            String substring = pcPassword.substring(j, j + 1);
+            int parseInt = Integer.parseInt(substring);
+            result[i] = (byte) ((parseInt & 0x0F) << 4);
+
+            substring = pcPassword.substring(j + 1, j + 2);
+            parseInt = Integer.parseInt(substring);
+            result[i] |= (byte) (parseInt & 0x0F);
+        }
+        return result;
+    }
+
+    public static String byteArrayAsString(byte[] array) {
+        if (array == null) {
+            return "";
+        }
+        return DatatypeConverter.printHexBinary(array);
```

Comment:
Please replace the usage of `DatatypeConverter.printHexBinary()` with the `HexUtils.bytesToHex()` method provided by the framework.

Reply:
Nice that we have it in the framework. I refactored all calls recursively and removed completely the method including subsequent calls.